tokio-postgres/src/config.rs

```diff
@@ -187,6 +222,7 @@ pub struct Config {
     pub(crate) ssl_mode: SslMode,
     pub(crate) ssl_root_cert: Option<Vec<u8>>,
     pub(crate) host: Vec<Host>,
+    pub(crate) hostaddr: Vec<IpAddr>,
```

Comment:
similar to the comment on the string parameters, it seems that it would be a lot cleaner to add a new enum type to the `Host` definition that tied together the mappings more clearly:

```
/// A host specification.
#[derive(Debug, Clone, PartialEq, Eq)]
pub enum Host {
    /// A TCP hostname.
    Tcp(String),
    /// A resolved TCP hostname and associated IP addresses to use.
    /// The hostname is only used when the SSL mode requires host verification.
    ResolvedTcp { host: String, ips: Vec<String> },
    /// A path to a directory containing the server's Unix socket.
    ///
    /// This variant is only available on Unix platforms.
    #[cfg(unix)]
    Unix(PathBuf),
}
```